### ISSUE TITLE
Problem with none installed dependencies

### ISSUE THREAD

**Issue by kirrg001** [TryGhost/Ghost-CLI#516]
See https://github.com/TryGhost/Ghost/commit/a7fe464d64bc1675c8ebc9e54b19e358dbcc8561 and https://github.com/nodejs/Release/issues/263.

Ghost requires Node 8.9 with 1.16.2. As soon as i run `ghost update --zip ...` with < 8.9, i'll get

> error ghost@1.16.2: The engine "node" is incompatible with this module. Expected version "^4.5.0 || ^6.9.0 || ^8.9.0".

Makes sense.

But when i run it another time and ignore it, it skips the check and continues. 

> ℹ Downloading and updating Ghost [skipped]

Because it skips this step, **no dependencies are installed** and my installation is in a broken state. With broken state i mean, the CLI tries to run migrations, Ghost requires modules and fails, because they are not installed.

```
ghost-local-1 │ /private/tmp/ghost-116     │ 1.16.2         │ stopped │ n/a │ n/a  │ n/a
```

I was on `1.16.1` before. Doesn't rollback to `1.16.1`.

Note: This can happen for every node incompatible error - not really related to the v8 bump.
Note: Maybe only related by using `--zip`?

**Comment by ErisDS** [comment]
`ghost update --rollback` should force the rollback to at least fix the install.

As a secondary aspect to this. I was planning to raise an issue today about designing a recommended process for upgrading Node with Ghost CLI, because, even without this problem, we need to figure out how this should be done.

Not sure if it's a documentation task only, or if we need some new tooling.

E.g. I am on Node v6.9, I want to be on the newer/faster/better Node v8. I upgrade Node on my machine, now what do I do to make Ghost compatible?

**Comment by kirrg001** [comment]
> ghost update --rollback should force the rollback to at least fix the install.

Yes it fixed it for me. Thanks 👍 Maybe worth adding this to our troubleshooting section.

**Comment by kirrg001** [comment]
> Ghost requires Node 8.9 with 1.16.2. As soon as i run ghost update --zip ... with < 8.9, i'll get
> error ghost@1.16.2: The engine "node" is incompatible with this module. Expected version "^4.5.0 || ^6.9.0 || ^8.9.0".

One extra note. The error was **only** in the CLI debug log. The instant error only showed that `yarn` failed. That might cause users to simply run `ghost update` again, because they are not looking directly into the debug log.

**Comment by pascalandy** [comment]
Hi Hannah,

I looked for the instructions over https://docs.ghost.org/docs/supported-node-versions but didn't find any workaround for Node-8.

I run these instructions to install Ghost

```
	ghost install "$GHOST_VERSION" \
		--db sqlite3 --no-prompt \
		--no-stack --no-setup \
		--dir "$GHOST_INSTALL"							&& \
	ghost config --ip 0.0.0.0 \
		--port 2368 --no-prompt --db sqlite3 \
		--url http://localhost:2368 \
		--dbpath "$GHOST_CONTENT/data/ghost.db"			&& \
	ghost config paths.contentPath "$GHOST_CONTENT"		;
```

Any workaround advice to share?
Thank you for everything!

**Comment by acburdine** [comment]
@pascalandy what exactly is the issue you're seeing? As long as you're on node 8.9, you should be fine.

There's a check going out in the next version of the CLI that should hopefully fix the original issue though.

**Comment by pascalandy** [comment]
Just found the issue.
I was running node `8.9.0` but I have to run `8.9.1`. It's all good now :)

**Comment by acburdine** [comment]
This should be fixed by #732 - if yarn fails to install we cleanup afterwards.
